packages/browser-destinations/destinations/google-analytics-4-web/src/__tests__/setConfigurationFields.test.ts

```diff
@@ -0,0 +1,177 @@
+    name: 'Set Configuration Fields',
+    enabled: true,
+    subscribe: 'type = "page"',
+    mapping: {
+      ads_storage_consent_state: {
+        '@path': '$.properties.ads_storage_consent_state'
+      },
+      analytics_storage_consent_state: {
+        '@path': '$.properties.analytics_storage_consent_state'
+      },
+      screen_resolution: {
+        '@path': '$.properties.screen_resolution'
+      },
+      user_id: {
+        '@path': '$.properties.user_id'
+      },
+      page_title: {
+        '@path': '$.properties.page_title'
+      },
+      page_referrer: {
+        '@path': '$.properties.page_referrer'
+      },
+      language: {
+        '@path': '$.properties.language'
+      },
+      content_group: {
+        '@path': '$.properties.content_group'
+      }
+    }
+  }
+]
+
+describe('Set Configuration Fields action', () => {
+  const settings: {
+    cookiePrefix?: undefined | string
+    enableConsentMode: boolean
+    measurementID: string
+    allowAdPersonalizationSignals: boolean
+    allowGoogleSignals: boolean
+    cookieDomain: string
+    cookieExpirationInSeconds: number
+    cookieUpdate: boolean
+    pageView: boolean
+    cookiePath?: string
+  } = {
+    enableConsentMode: false,
+    measurementID: 'G-XXXXXXXXXX',
+    allowAdPersonalizationSignals: false,
+    allowGoogleSignals: false,
+    cookieDomain: 'auto',
+    cookieExpirationInSeconds: 63072000,
+    cookieUpdate: true,
+    pageView: true
+  }
+  beforeEach(async () => {
+    jest.restoreAllMocks()
+
+    const [trackEventPlugin] = await googleAnalytics4Web({
+      ...settings,
+      subscriptions
+    })
+    setConfigurationEvent = trackEventPlugin
+
+    jest.spyOn(destination, 'initialize').mockImplementation(() => {
+      mockGtag = jest.fn()
+      return Promise.resolve(mockGtag)
+    })
+    await trackEventPlugin.load(Context.system(), {} as Analytics)
+  })
+
+  it('should call gtag with the correct config when consent mode is enabled', async () => {
+    settings.enableConsentMode = true
+
+    const [trackEventPlugin] = await googleAnalytics4Web({
+      ...settings,
+      subscriptions
+    })
+    setConfigurationEvent = trackEventPlugin
+    await trackEventPlugin.load(Context.system(), {} as Analytics)
+
+    const context = new Context({
+      event: 'setConfigurationFields',
+      type: 'page',
+      properties: {
+        ads_storage_consent_state: 'granted',
+        analytics_storage_consent_state: 'denied'
+      }
+    })
+
+    setConfigurationEvent.page?.(context)
+
+    expect(mockGtag).toHaveBeenCalledWith('consent', 'update', {
+      ad_storage: 'granted',
+      analytics_storage: 'denied'
+    })
+    expect(mockGtag).toHaveBeenCalledWith('config', 'G-XXXXXXXXXX', {
+      allow_ad_personalization_signals: false,
+      allow_google_signals: false
+    })
+  })
+  it('should call gtag with the correct config when consent mode is disabled and having values in payload', async () => {
```

Comment:
Does this name make sense?

```suggestion
  it('should not configure consent when consent mode is disabled', async () => {
```